python/dlisio/load.py

```diff
@@ -88,88 +88,209 @@ def load(path, error_handler = None):
     if not error_handler:
         error_handler = ErrorHandler()
 
-    sulsize = 80
-    tifsize = 12
-    lfs = []
-
-    def rewind(offset, tif):
-        """Rewind offset to make sure not to miss VRL when calling findvrl"""
-        offset -= 4
-        if tif: offset -= 12
-        return offset
-
     path = str(path)
     if not os.path.isfile(path):
         raise OSError("'{}' is not an existing regular file".format(path))
+
     stream = open(path)
+    tm = core.read_tapemark(stream)
+    is_tif = core.valid_tapemark(tm)
+    stream.close()
+
+    # Processing logic:
+    # - Open cfile/tif protocol at the start of every logical file.
+    #   Note that TIF protocol may report EOF before physical file is EOF, thus
+    #   it's important to open it anew every time.
+    # - Read SUL if it is found.
+    # - Parse Logical file.
+    # - As TIF EOF marks can happen before SUL, before LF or before physical
+    #   EOF, test for EOF before attempting to read SUL or LF blocks.
+    # - Close streams that do not contain valid Logical Files.
+    # - Attempt to open logical files until physical file reports data end.
```

Comment:
iho there are only a couple of things that are unclear about the while loop:

1) what's the deal with the `close + continue`
2) Where is what protocol opened
3) That parser.eof() also is true for truncated files

I would remove the current comments and add an inline comment about 1), properly explaining that this is a tif thing. As pr other comments I would address 2) by explicitly open rp66 in the loop (not hide it in parse_LF, and maybe rename open_stream. And point 3) by splitting EOF and truncated into separate attributes

Reply:
1. I feel like adding inline comments here spoil the flow too much, so I moved the explanation to logical_eof docs?
2. rp66 is called explicitly, stream_open name left as is because nothing better comes to mind :D 
3. eof stuff discussed